components/post_view/post_image.jsx

```diff
@@ -99,14 +103,10 @@ export default class PostImageEmbed extends React.PureComponent {
 
     render() {
         if (this.state.errored || !this.state.loaded) {
+            // scroll pop could be improved with better placeholder during !this.  state.loading
             return null;
         }
-
-        var url = this.props.link;
-        if (global.window.mm_config.HasImageProxy === 'true') {
-            url = Client4.getBaseRoute() + '/image?url=' + encodeURIComponent(url);
-        }
-
+        const url = this.imagesAreProxied ? this.proxyEncodeImageUrl(this.props.link) : this.props.link;
```

Comment:
Could you maybe move `imagesAreProxied` into this function? The name makes it more clear what's going on here, but having it set in the constructor to only be used here is a bit confusing

Reply:
I put it in the constructor since it should never change and thus is set the one time. Although it is such a simple check I will move it into the render for clarity